File: Programmers/[12938] 최고의 집합/ONE/12938.java
class Solution {
    public int[] solution(int n, int s) {
        if (n > s) {
            return new int[]{-1};
        }

        int[] answer = new int[n];
        int basic = s / n;
        int remain = s % n;

        for (int index = 0; index < n; index++) {
            answer[index] = basic;
            if (index >= n - remain) {
                answer[index]++;
            }
        }

        return answer;
    }
}

File: Programmers/[12938] 최고의 집합/KingDonggyu/12938.js
function solution(n, s) {
  if (n > s) {
    return [-1];
  }

  const answer = [];

  while (n) {
    const value = Math.floor(s / n);
    answer.push(value);
    s -= value;
    n--;
  }

  return answer;
}
